Xamarin.PropertyEditing.Tests/AsyncWorkQueueTests.cs

```diff
@@ -47,6 +47,24 @@ public void Setup ()
 		}
 
 		[Test, Timeout (1000)]
+		public async Task SameRequesterAfterNonSameIsFreed ()
```

Comment:
It's unclear to me what this is testing, and how it would fail.

Reply:
The expanded commit message goes into the details, this test fails on the previous implementation. The trick is that the work queue allows items that are from the same requester as the active item to immediately begin executing. The prior implementation only had this pass-through working when it was _already_ the active item, but not when it became it _later_.